src/lib/validate.js

```diff
@@ -9,7 +9,7 @@ module.exports.validateUsernameLocally = username => {
         return {valid: false, errMsgId: 'registration.validationUsernameMinLength'};
     } else if (username.length > 20) {
         return {valid: false, errMsgId: 'registration.validationUsernameMaxLength'};
-    } else if (!/^[\w-]+$/i.test(username)) {
+    } else if (!/^[\w_-]+$/i.test(username)) {
```

Comment:
I thought \w was supposed to include underscores? I just created dinopickles_test through the new join flow...

Reply:
aaaahhhhh you're right, OK, that makes it not so surprising that we left it like this!!!!!!

Took that change out, but including the test additions/revisions because I think those are good anyway